lib/practice_exercises.rb

```diff
@@ -1,13 +1,53 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(n)
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  word_compare = ""
+  regex_string = ""
+  i = 0 
+
+  # string = [racecar1]
+  while i < string.length  do 
+    if string[-(i + 1)][/[a-zA-Z0-9]/]
+      word_compare += string[-(i + 1)].downcase 
+    end  
+    if string[i][/[a-zA-Z0-9]/]
+      regex_string += string[i].downcase
+    end
+    i += 1 
+  end
+  
+  return word_compare == regex_string
+    
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(n)
 def longest_prefix(strings)
```

Comment:
I would say the time complexity is O( (n log n) +  m) where m is the length of the shortest word and n being the number of strings.  Otherwise 👍